torch/csrc/jit/ir.h

```diff
@@ -456,7 +463,9 @@ struct Node : public Attributes<Node> {
   // concrete type as 'this', but in graph 'g' which might be different
   // than graph_
   virtual Node * allocNewInstance(Graph * g) {
-    return new Node(g, kind());
+    Node * ret = new Node(g, kind());
+    ret->setPrettyName(prettyName());
```

Comment:
I think this isn't supposed to copy any node attributes. This function is needed only because you can't have virtual constructors in C++, so you need to use it to allocate a "clone" of a node that has appropriate runtime info and memory layout

Reply:
hmm. Ok, I'll take a closer look here